### DISCUSSION TITLE
Image control and HDR rendering

### DISCUSSION THREAD

**Discussion by Gavin-Williams** [microsoft/microsoft-ui-xaml#8662] [Q&A]
I see that the image control supports jxr (jpg xr). Does the image control display a 16 bit rendertarget? Or does it get remapped to 8 bits. There is an open issue that WinUI doesn't support HDR https://github.com/Microsoft/microsoft-ui-xaml/issues/67 so I'm not sure if that issue affects the image control. 

Also, do you see jxr as an alternative to exr and maybe even dds? I haven't gone over all the features, but jxr supports up to 32 bit color channels, an alpha channel, it has both lossy and lossless compression. It has tiled storage. So it seems to have broad utility, but not sure if it's quite as flexible as exr.

Update, looking at bmp, I didn't know I could use it for 16/32 bit images. How does that display in the image control ... correctly?

**Comment by nbryant42** [comment]
I know it's an old thread, but to save somebody else the trouble: I just tested plugging the Image control into my own WinUI 3 XAML with my own photography:

```xml
<Image Source="/Assets/GrandTeton.jxr"/>
```

Sadly, this does not work; the result is clipped to SDR. Maybe this worked in UWP (but I've never tried building anything in UWP.)

This doc seems to imply that MS is moving towards similar functionality implemented directly on the Visual Layer: https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.media.loadedimagesurface?view=windows-app-sdk-1.7

To this question: "jxr as an alternative to exr and maybe even dds?". These days, one of the best ways forward (in a browser environment, at least) is a JPEG with gain map, which can adapt to SDR or HDR end-user monitors quite well if you do a good job with the encoding. [Greg Benz' website](https://gregbenzphotography.com/hdr/) is quickly becoming the foremost authority on the subject.

**Reply by Gavin-Williams** [reply]
Old thread but still very relevant. Thanks for the updated information. I don't think it would have worked in UWP. HDR (and WCG) support is evolving still, and UWP seems like too long ago to me.

**Reply by nbryant42** [reply]
I've been doing a bit of a deep dive on this. The way to get HDR rendering to work is via `SwapChainPanel`. [This sample code](https://github.com/microsoft/Windows-universal-samples/tree/main/Samples/D2DAdvancedColorImages) shows how, and I was able to take a similar approach in one of my small projects (documenting some gotchas I ran into in some [Developer Notes](https://github.com/nbryant42/xDRCal?tab=readme-ov-file#developer-notes))
